libgstc/c/libgstc.c

```diff
@@ -983,3 +987,146 @@ gstc_pipeline_list (GstClient * client, char **pipelines[], int *list_lenght)
 out:
   return ret;
 }
+
+GstcStatus
+gstc_pipeline_list_signals (GstClient * client, const char *pipeline_name,
+    const char *element, char **signals[], int *list_lenght)
+{
+  GstcStatus ret;
+  char *what;
+  int asprintf_ret;
+  char *response;
+
+  gstc_assert_and_ret_val (NULL != client, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != pipeline_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != element, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != signals, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != list_lenght, GSTC_NULL_ARGUMENT);
+
+  asprintf_ret =
+      asprintf (&what, PIPELINE_SIGNAL_LIST_FORMAT, pipeline_name, element);
+  if (PRINTF_ERROR == asprintf_ret) {
+    return GSTC_OOM;
```

Comment:
No, since `what` isn't actually initialized after this function there is no need to free it in `out`. This is the standard followed in the other functions, see for example `get_graph`